## Essentials of Statistics and Math for Data Science - Assignment 3
Questions 1: A biologist wants to study the relationship between the size of a plant and its growth rate. They have collected data on the size and growth rate of 50 plants and want to use least squares regression to estimate the relationship between these variables. what is the 95% confidence interval for the slope of the regression line?
solve it.
Size (x):
[10, 12, 14, 16, 18, 20, 22, 24, 26, 28, 30, 32, 34, 36, 38, 40, 42, 44, 46, 48, 50, 52, 54, 56, 58, 60, 62, 64, 66, 68, 70, 72, 74, 76, 78, 80, 82, 84, 86, 88, 90, 92, 94, 96, 98, 100, 102, 104, 106, 108]
Growth rate (y):
[6.8, 8.148,9.496,10.844 , 12.192 ,13.540 , 14.888 , 16.236 , 17.584 , 18.932 ,  20.280 , 21.628 , 22.976 , 24.324 , 25.672 , 27.020 , 28.368 , 29.716 , 31.064 , 32.412 , 33.760 , 35.108 , 36.456 , 37.804 , 39.152 , 40.5 , 41.848 , 43.196 , 44.544 , 45.892 , 47.240 , 48.588 , 49.936 , 51.284 , 52.632 , 53.980 , 55.328 , 56.676 , 58.024 , 59.372 , 60.720 , 62.068 , 63.416 ,  64.764 , 66.112 , 67.460 , 68.808 , 70.156 , 71.504 , 72.852 ]

In [38]:
# Import required libraries
import numpy as np
import statsmodels.api as sm
import pandas as pd

In [39]:
# Data: Size (x) and Growth rate (y)
x = np.array([10, 12, 14, 16, 18, 20, 22, 24, 26, 28, 30, 32, 34, 36, 38, 40, 42, 44, 46, 48, 50, 52, 54, 56, 58, 60, 62, 64, 66, 68, 70, 72, 74, 76, 78, 80, 82, 84, 86, 88, 90, 92, 94, 96, 98, 100, 102, 104, 106, 108])
y = np.array([6.8, 8.148, 9.496, 10.844, 12.192, 13.540, 14.888, 16.236, 17.584, 18.932, 20.280, 21.628, 22.976, 24.324, 25.672, 27.020, 28.368, 29.716, 31.064, 32.412, 33.760, 35.108, 36.456, 37.804, 39.152, 40.5, 41.848, 43.196, 44.544, 45.892, 47.240, 48.588, 49.936, 51.284, 52.632, 53.980, 55.328, 56.676, 58.024, 59.372, 60.720, 62.068, 63.416, 64.764, 66.112, 67.460, 68.808, 70.156, 71.504, 72.852])

In [40]:
# Add a constant to the independent variable matrix (intercept)
X = sm.add_constant(x)

In [41]:
# Fit the model
model = sm.OLS(y, X).fit()

In [42]:
# Get the summary
summary = model.summary()

In [43]:
# Extract the 95% confidence interval for the slope
conf_int = model.conf_int(alpha=0.05)

In [44]:
# Output the results
print("Model Summary:\n", summary)
print("\n95% Confidence Interval for the Slope:\n", conf_int)

Model Summary:
                             OLS Regression Results                            
Dep. Variable:                      y   R-squared:                       1.000
Model:                            OLS   Adj. R-squared:                  1.000
Method:                 Least Squares   F-statistic:                 5.674e+32
Date:                Sat, 27 Jul 2024   Prob (F-statistic):               0.00
Time:                        14:49:16   Log-Likelihood:                 1569.3
No. Observations:                  50   AIC:                            -3135.
Df Residuals:                      48   BIC:                            -3131.
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const          0.0600   1.86e-15   3

#### Questions 2: Suppose we want to test whether the mean height of students in a particular school is different from 168 cm. We collect a random sample of 25 students from the school and measure their heights. The sample mean height is 170 cm and the sample standard deviation is 5 cm. Conduct a hypothesis test at the 5% significance level to determine whether there is evidence to suggest that the mean height of students in the school is different from 168 cm.

In [9]:
import scipy.stats as stats

sample_mean = 170
population_mean = 168
sample_std_dev = 5
sample_size = 25

In [10]:
# Calculate the test statistic (t)
t_statistic = (sample_mean - population_mean) / (sample_std_dev / (sample_size ** 0.5))

In [11]:
# Calculate the critical value for a two-tailed test
alpha = 0.05
df = sample_size - 1
critical_value = stats.t.ppf(1 - alpha / 2, df)

In [12]:
# Calculate the p-value
p_value = 2 * (1 - stats.t.cdf(abs(t_statistic), df))

In [13]:
# Output the results
print("Test Statistic (t):", t_statistic)
print("Critical Value (t) for 5% significance level:", critical_value)
print("p-value:", p_value)

Test Statistic (t): 2.0
Critical Value (t) for 5% significance level: 2.0638985616280205
p-value: 0.056939849936591624


In [14]:
# Decision
if abs(t_statistic) > critical_value:
    print("Reject the null hypothesis: There is evidence to suggest that the mean height is different from 168 cm.")
else:
    print("Fail to reject the null hypothesis: There is not enough evidence to suggest that the mean height is different from 168 cm.")

Fail to reject the null hypothesis: There is not enough evidence to suggest that the mean height is different from 168 cm.


#### Questions 3: A hospital has developed a diagnostic test to detect a certain disease. The test has been found to have a false positive rate of 5%, meaning that 5% of people who do not have the disease will test positive for it. Similarly, the test has a false negative rate of 2%, meaning that 2% of people who do have the disease will test negative for it. If the disease has a prevalence rate of 2% in the population, what is the probability that a patient who tests positive has the disease?

In [16]:
false_positive_rate = 0.05
false_negative_rate = 0.02
prevalence_rate = 0.02

In [17]:
# True Positive Rate
true_positive_rate = 1 - false_negative_rate

In [18]:
# Probability of Testing Positive
prob_positive_test = (true_positive_rate * prevalence_rate) + (false_positive_rate * (1 - prevalence_rate))

In [19]:
# Probability of Disease Given Positive Test
prob_disease_given_positive = (true_positive_rate * prevalence_rate) / prob_positive_test

prob_disease_given_positive

0.28571428571428575

#### Questions 4: A car rental company claims that the average fuel efficiency of its fleet of cars is 30 miles per gallon (mpg). To test this claim, a random sample of 25 cars from the fleet is selected, and their fuel efficiencies are recorded. The sample has a mean fuel efficiency of 28.5 mpg with a standard deviation of 3.2 mpg. Test the company's claim using a significance level of 0.05.

In [20]:
import scipy.stats as stats
import numpy as np

sample_mean = 28.5
population_mean = 30
sample_std = 3.2
sample_size = 25
alpha = 0.05

In [26]:
# Calculate the test statistic
t_statistic = (sample_mean - population_mean) / (sample_std / np.sqrt(sample_size))
t_statistic

-2.34375

In [27]:
# Degrees of freedom
df = sample_size - 1
df

24

In [28]:
# Critical t-value for two-tailed test
t_critical = stats.t.ppf(1 - alpha / 2, df)
t_critical

2.0638985616280205

In [29]:
# Calculate the p-value
p_value = 2 * (1 - stats.t.cdf(np.abs(t_statistic), df))
p_value

0.027705144868591436

In [31]:
t_statistic, t_critical, p_value

(-2.34375, 2.0638985616280205, 0.027705144868591436)

Conclusion:

The calculated test statistic is -2.34, which is less than t_critical -2.064.
The p-value is 0.0277, which is less than 0.05.
Therefore, we reject the null hypothesis. There is sufficient evidence at the 5% significance level to conclude that the average fuel efficiency of the fleet is different from 30 mpg.

#### Questions 5: A company wants to determine if there is a significant difference in the mean annual salaries of its employees between two departments: Department A and Department B. A random sample of 30 employees from Department A has a mean annual salary of $55,000 with a standard deviation of $5,000, while a random sample of 35 employees from Department B has a mean annual salary of $57,000 with a standard deviation of $4,500. Test the hypothesis using a significance level of 0.01.

To determine if there is a significant difference in the mean annual salaries between Department A and Department B, we will conduct a two-sample t-test. This test compares the means of two independent samples to see if there is a statistically significant difference between them.

Null Hypothesis (H₀): There is no significant difference in the mean salaries of employees between the two departments.
 
Alternative Hypothesis (H₁): There is a significant difference in the mean salaries of employees between the two departments.

In [32]:
# Department A
mean_A = 55000
std_A = 5000
n_A = 30

# Given values for Department B
mean_B = 57000
std_B = 4500
n_B = 35

In [33]:
# Calculate the test statistic
t_statistic = (mean_A - mean_B) / np.sqrt((std_A**2 / n_A) + (std_B**2 / n_B))

In [34]:
# Calculate degrees of freedom using the formula for unequal variances
df_numerator = ((std_A**2 / n_A) + (std_B**2 / n_B))**2
df_denominator = ((std_A**2 / n_A)**2 / (n_A - 1)) + ((std_B**2 / n_B)**2 / (n_B - 1))
df = df_numerator / df_denominator

In [35]:
# Calculate the critical t-value for a two-tailed test
t_critical = stats.t.ppf(1 - 0.01 / 2, df)

In [36]:
# Calculate the p-value
p_value = 2 * (1 - stats.t.cdf(np.abs(t_statistic), df))

In [37]:
t_statistic, t_critical, p_value

(-1.6831673346877503, 2.661769465401159, 0.09762827207415348)

Results
Test Statistic (t): -1.68
Critical t-Value (t₀.₀₁): 2.66
P-Value: 0.097

Conclusion:

Since the absolute value of the test statistic -1.68 is less than the critical value  2.66 or since the p-value 0.097 is greater than 0.01, we do not reject the null hypothesis.
Based on these calculations, there is no significant difference in the mean salaries of employees between the two departments.